### PR TITLE
Adding Mime warning to fsck

### PR DESCRIPTION
**Pull request by AnomalRoil**
RELEASE_NOTES=[ENHANCEMENT] fsck now detects leftover Mime secrets
RELEASE_NOTES=[BUGFIX] fsck won't correct recipients without --decrypt

### CODE REVIEW

File: internal/store/leaf/fsck.go [hunk 1]
```diff
@@ -59,10 +62,14 @@ func (s *Store) fsckCheckEntry(ctx context.Context, name string) error {
 	// make sure we can actually decode this secret
 	// if this fails there is no way we could fix this
 	if IsFsckDecrypt(ctx) {
-		_, err := s.Get(ctx, name)
+		ctx = ctxutil.WithShowParsing(ctx, true)
+		secret, err := s.Get(ctx, name)
 		if err != nil {
 			return errors.Wrapf(err, "failed to decode secret %s: %s", name, err)
 		}
+		if secret.Type() == "Mime" {
```
Comment (dominikschulz) [inline internal/store/leaf/fsck.go:70]: This should be: `if kv, ok := secret.(secrets.KV); ok && kv.FromMime() {`

See my other comment.

**Review by dominikschulz — CHANGES_REQUESTED** [review]
I like the feature, but please change the implementation.
What you're doing looks like poor-mans type checking.
Go has built-in support for that. That would be cleaner and also more concise.

Please:
- keep fsck.go mostly as-is (except for my comment there)
- change kv.go for have a func FromMime() bool
- Remove all other changes

**Comment by AnomalRoil** [comment]
I should have thought of using type assertions, sorry for that.
Done.

**Review by dominikschulz — APPROVED** [review]
LGTM

File: internal/store/leaf/fsck.go [hunk 2]
```diff
@@ -59,10 +63,15 @@ func (s *Store) fsckCheckEntry(ctx context.Context, name string) error {
...
 	// if this fails there is no way we could fix this
 	if IsFsckDecrypt(ctx) {
-		_, err := s.Get(ctx, name)
+		// we need to make sure Parsing is enabled in order to parse old Mime secrets
+		ctx = ctxutil.WithShowParsing(ctx, true)
+		secret, err := s.Get(ctx, name)
 		if err != nil {
 			return errors.Wrapf(err, "failed to decode secret %s: %s", name, err)
 		}
+		if kv, ok := secret.(*secrets.KV); ok && kv.FromMime() {
```
Comment (dominikschulz) [inline internal/store/leaf/fsck.go:72]: Actually we could have type checked against a locally defined interface `type fromMime interface { FromMime() bool }` here, but let's leave it as is for now.

**Pull request merged**